test/unit/mdc-tabs/mdc-tab-bar.test.js

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assert} from 'chai';
+import bel from 'bel';
+import domEvents from 'dom-events';
+import td from 'testdouble';
+import {createMockRaf} from '../helpers/raf';
+import {supportsCssVariables} from '../../../packages/mdc-ripple/util';
+import {MDCTab} from '../../../packages/mdc-tabs/tab';
+import {MDCTabBar} from '../../../packages/mdc-tabs/tab-bar';
+import {MDCTabBarFoundation} from '../../../packages/mdc-tabs/tab-bar';
+import {cssClasses} from '../../../packages/mdc-tabs/tab/constants';
+
+function getFixture() {
+  return bel`
+    <div>
+      <nav id="basic-tabs" class="mdc-tab-bar">
+        <a class="mdc-tab mdc-tab--active" href="#one">Item One</a>
+        <a class="mdc-tab" href="#two">Item Two</a>
+        <a class="mdc-tab" href="#three">Three</a>
+        <span class="mdc-tab-bar__indicator"></span>
+      </nav>
+    </div>`;
+}
+
+function setupTest() {
+  const fixture = getFixture();
+  const root = fixture.querySelector('.mdc-tab-bar');
+  const indicator = fixture.querySelector('.mdc-tab-bar__indicator');
+  const component = new MDCTabBar(root);
```

Comment:
It doesn't look like you're mocking out tabs here. This means that tab bar is not being tested under _isolation_. Bugs in `MDCTab` may cause `MDCTabBar` to fail. I'd suggest using `tabFactory` to create Mock `MDCTab` instances that are guaranteed to work. See https://github.com/material-components/material-components-web/blob/master/test/unit/mdc-select/mdc-select.test.js#L24-L63 for an example